spec/griddler/amazon_ses_spec.rb

```diff
@@ -43,6 +43,24 @@
     it 'should return the text body in UTF-8' do
       expect(Griddler::AmazonS3SES::Adapter.normalize_params(default_params)[:text].encoding.to_s).to eq "UTF-8"
     end
+
+    it 'should return the text body content for non-multi-part email' do
+      message = Mail.read_from_string(mail_content)
+      allow_any_instance_of(Griddler::AmazonS3SES::Adapter).to receive(:message).and_return(message)
+      expect(Griddler::AmazonS3SES::Adapter.normalize_params(default_params)[:text]&.strip).to eq "Hi"
+    end
+
+    it 'should return the text body content for multi-part email' do
+      message = Mail.read_from_string(mail_content_multipart_text_and_html)
+      allow_any_instance_of(Griddler::AmazonS3SES::Adapter).to receive(:message).and_return(message)
+      expect(Griddler::AmazonS3SES::Adapter.normalize_params(default_params)[:text]&.strip).to eq "Hello! I am a multi-part text."
+    end
+
+    it 'should return empty string for multi-part email with no text content' do
+      message = Mail.read_from_string(mail_content_multipart_html_only)
+      allow_any_instance_of(Griddler::AmazonS3SES::Adapter).to receive(:message).and_return(message)
+      expect(Griddler::AmazonS3SES::Adapter.normalize_params(default_params)[:text]&.strip).to eq ""
+    end
```

Comment:
this test case would crash with the frozen string error without the fix in this PR.

Run rspec with:
```
rspec spec/griddler/amazon_ses_spec.rb
```

Reply:
Thank you for adding these tests!